### PR TITLE
fix(components): [result] rename slot's name from 'subTitle' to 'sub-title'

### LINKED ISSUE #6528: [Bug Report] [Component] [result] Result 组件 subTitle 插槽不生效
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.31`
- Element Plus Version: `2.0.5`
- Browser / OS: `UserAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36 Edg/98.0.1108.56`
- Build Tool: `Webpack`

### Related Component

- `el-result`

### Reproduction Link

[CodePen](https://codepen.io/lonquan/pen/vYWoVBB)

### Steps to reproduce

<el-result icon="error" sub-title="sub-title-prop">
        <template #sub-title>
            sub-title
        </template>
        <template #title>
            <span>Title</span>
        </template>
        <template #subTitle>
            subTitle
        </template>
        <template #extra>
            <el-button type="primary" auto-insert-space>返回</el-button>
        </template>
    </el-result>

### What is Expected?

正确使用 subTitle 插槽并且生效

### What is actually happening?

subTitle 插槽不生效

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by freedomlang**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

fix #6528 the slot name ends up in the attribute name, so it is normalized to all-lowercase by the browser. Rename it to lowercase to avoid mismatch.

### CODE REVIEW

File: packages/components/carousel/src/main.vue [hunk 1]
```diff
@@ -357,6 +357,17 @@ export default defineComponent({
...
+        () => items.value,
+        (val) => {
+          if (val.length) {
+            playSlides()
+            stopWatchItems?.()
+          }
+        },
+        { immediate: true, deep: true }
+      )
+
```
Comment (JeremyWuuuuu) [inline packages/components/carousel/src/main.vue:370]: @freedomlang Did you accidentally included code which doesn't belong to this issue? Would you please remove them?

**Comment by JeremyWuuuuu** [comment]
@freedomlang Code seems legit, please remove the undesired code

File: packages/components/carousel/src/main.vue (hunk 1 above)
Comment (freedomlang) [reply packages/components/carousel/src/main.vue]: Removed

**Review by jw-foss — APPROVED** [review]

**Review by kooriookami — APPROVED** [review]

**Comment by chenxch** [comment]
@JeremyWuuuuu Does this change make it unusable after the upgrade?

**Comment by JeremyWuuuuu** [comment]
@chenxch I think essentially we should keep `sub-title` way. Could be considered as a minor breaking change, we might warn the users.

**Comment by JeremyWuuuuu** [comment]
I'll make a PR for warning user who is still using this attribute.

**Comment by chenxch** [comment]
@JeremyWuuuuu This is a good direction, get is also helpful for web components.

**Comment by JeremyWuuuuu** [comment]
@chenxch https://github.com/element-plus/element-plus/pull/6651 Added, when user using `subTitle` as slot in dev mode they will get a warning message.

**Pull request merged**